code/modules/research/xenobiology/xenobiology_dissections.dm

```diff
@@ -0,0 +1,1533 @@
+		revert_cast()
+		return
+	if(get_dist(target.loc, user.loc) > 2)
+		to_chat(user, "<span class = 'warning'>They're too far away!</span>")
+		revert_cast()
+	if(!length(get_path_to(user, target, max_distance = 2, simulated_only = FALSE, skip_first = FALSE)))
+		to_chat(user, "<span class='warning'>Something is blocking the way!</span>")
+		revert_cast()
+		return
+	if(quality == ORGAN_BROKEN)
+		if(prob(25))
+			targets += user
+			user.adjustBruteLoss(10)
+			user.adjustFireLoss(10)
+			to_chat(user, "<span class = 'danger'>You get dragged along into bluespace, your flesh searing from the unstable energies!</span>")
+		else
+			to_chat(user, "<span class = 'danger'>Drawing upon unstable energy singes your flesh!</span>")
+			user.adjustFireLoss(8)
+	user.mob_light(LIGHT_COLOR_PURPLE, 3, _duration = 3)
+	new /obj/effect/temp_visual/hierophant/telegraph/teleport(get_turf(target))
+	. = ..()
+
+/obj/item/organ/internal/kidneys/xenobiology/shivering
+	name = "Shivering Organ"
+	desc = "It constantly shivers, seeking to warm itself from its environment."
+	analyzer_price = 30
+
+/obj/item/organ/internal/kidneys/xenobiology/shivering/on_life()
+	. = ..()
+	if(owner.get_temperature() < owner.dna.species.cold_level_1 + 40)
+		switch(organ_quality)
+			if(ORGAN_DAMAGED)
+				owner.bodytemperature += 15
+			if(ORGAN_NORMAL)
+				owner.bodytemperature += 20
+			if(ORGAN_PRISTINE)
+				owner.bodytemperature += 30
+
+/obj/item/organ/internal/kidneys/xenobiology/sweating
+	name = "Sweaty Organ"
+	desc = "It constantly sweats, seeking to cool itself off from its environment."
+	analyzer_price = 30
+	hidden_origin_tech = TECH_TOXINS
+	hidden_tech_level = 6
+
+/obj/item/organ/internal/kidneys/xenobiology/shivering/on_life()
+	. = ..()
+	if(owner.get_temperature() > owner.dna.species.heat_level_1 - 40)
+		switch(organ_quality)
+			if(ORGAN_DAMAGED)
+				owner.bodytemperature -= 15
+			if(ORGAN_NORMAL)
+				owner.bodytemperature -= 20
+			if(ORGAN_PRISTINE)
+				owner.bodytemperature -= 30
+
+/obj/item/organ/internal/liver/xenobiology/soupy
+	name = "Soupy Organ"
+	desc = "This organ seems to barely keep its own form together. It also reeks of tomato sauce."
+	analyzer_price = 15
+	var/original_name
+	var/original_own_species_blood
+	var/original_exotic_blood
+	var/original_blood_color
+	var/original_blood_type
+	hidden_origin_tech = TECH_BIO
+	hidden_tech_level = 6
+
+/obj/item/organ/internal/liver/xenobiology/soupy/insert(mob/living/carbon/human/M, special = 0, dont_remove_slot = 0)
+	. = ..()
+	original_name = M.dna.species.name
+	original_own_species_blood = M.dna.species.own_species_blood
+	original_exotic_blood = M.dna.species.exotic_blood
+	original_blood_color = M.dna.species.blood_color
+	original_blood_type = M.dna.blood_type
+
+	M.dna.species.name = "Tomato Hybrid"
+	M.dna.species.own_species_blood = TRUE
+	M.dna.species.exotic_blood = "tomatojuice"
+	M.dna.species.blood_color = "#e25821"
+	M.dna.blood_type = "Tomato"
```

Comment:
this is terrifying. will people with this be able to get blood back by eating tomatoes, or are they SOL?